src/worker/lambda/handlerPuller.go

```diff
@@ -1,294 +1,154 @@
-			if err != nil {
-				return err
-			}
-
-			relPath, err := filepath.Rel(src, path)
-			if err != nil {
-				return err
-			}
+func NewHandlerPuller(dirMaker *common.DirMaker) (*HandlerPuller, error) {
+	ctx := context.Background()
+	storeURL := common.Conf.Registry
 
-			if d.IsDir() {
-				dirInfo, err := d.Info()
-				if err != nil {
-					return err
-				}
-				return os.MkdirAll(filepath.Join(dest, relPath), dirInfo.Mode())
-			}
-
-			return copyFile(path, filepath.Join(dest, relPath))
-		})
-	}
-	return copyFile(src, dest)
-}
-
-func copyFile(src, dest string) error {
-	srcFile, err := os.OpenFile(src, os.O_RDWR|os.O_CREATE|syscall.O_NOFOLLOW, 0666)
-	if err != nil {
-		return err
+	// If no recognized scheme is present, assume local path and add file://
+	if !strings.HasPrefix(storeURL, "file://") &&
+		!strings.HasPrefix(storeURL, "s3://") &&
+		!strings.HasPrefix(storeURL, "gs://") {
+		storeURL = "file://" + storeURL
 	}
-	defer srcFile.Close()
 
-	// Get source file permissions
-	srcInfo, err := srcFile.Stat()
-	if err != nil {
-		return err
+	if strings.HasPrefix(storeURL, "file://") {
+		dir := strings.TrimPrefix(storeURL, "file://")
+		if err := os.MkdirAll(dir, 0755); err != nil {
+			return nil, fmt.Errorf("failed to create local lambda store directory: %w", err)
+		}
 	}
 
-	destFile, err := os.Create(dest)
-	if err != nil {
-		return err
-	}
-	defer destFile.Close()
+	bucket, err := blob.OpenBucket(ctx, storeURL)
 
-	_, err = io.Copy(destFile, srcFile)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("failed to open blob bucket: %w", err)
 	}
 
-	// Set destination file permissions to match source
-	return destFile.Chmod(srcInfo.Mode())
-}
-
-func NewHandlerPuller(dirMaker *common.DirMaker) (cp *HandlerPuller, err error) {
 	return &HandlerPuller{
-		prefix:   common.Conf.Registry,
+		bucket:   bucket,
 		dirMaker: dirMaker,
 	}, nil
 }
 
-func (cp *HandlerPuller) isRemote() bool {
-	return strings.HasPrefix(cp.prefix, "http://") || strings.HasPrefix(cp.prefix, "https://")
-}
-
-func (cp *HandlerPuller) Pull(name string) (rt_type common.RuntimeType, targetDir string, err error) {
+func (cp *HandlerPuller) Pull(name string) (common.RuntimeType, string, error) {
 	t := common.T0("pull-lambda")
 	defer t.T1()
 
 	if err := common.ValidateFunctionName(name); err != nil {
-		return rt_type, "", err
+		return RT_UNKNOWN, "", err
 	}
 
-	if cp.isRemote() {
-		// registry type = web - only support tar.gz files
-		urls := []string{
-			cp.prefix + "/" + name + ".tar.gz",
-		}
+	key := name + ".tar.gz"
 
-		for i := 0; i < len(urls); i++ {
-			rt_type, targetDir, err = cp.pullRemoteFile(urls[i], name)
-			if err == nil {
-				return rt_type, targetDir, nil
-			} else if err != errNotFound404 {
-				// 404 is OK, because we just go on to check the next URLs
-				return rt_type, "", err
-			}
+	attrs, err := cp.bucket.Attributes(context.Background(), key)
+	if err == nil {
+		version := attrs.ModTime.String()
```

Comment:
Why convert the ModTime to a string, instead of just storing the version as a timestamp?  Do you envision it being represented differently at some point?